webapp/src/http_handlers/sessions/sessions_tracker.rs

```diff
@@ -61,45 +87,108 @@ impl SessionsTracker
         }
     }
 
-    /// Builder method, that sets sessions life duration
+    /// ## Builder method, that sets sessions life duration.
+    /// ----
+    /// Args:
+    /// ---
+    /// * `duration` - new `life_duration` value.
     pub fn life_duration(mut self, duration: &Duration) -> Self
     {
         self.life_duration = duration.to_owned();
         self
     }
 
-    /// Returns a ref to a tracked session by its id
-    pub fn get_session(&self, id: &u128) -> Option<&Session>
+    /// ## Locks `sessions_data`.
+    /// ----
+    /// ## Returns:
+    /// `MutexGuard` for `sessions`. If the mutex is poisoned - it will end the process with the exit code of 1.
+    fn lock(&self) -> MutexGuard<HashMap<u128, Session>>
     {
-        self.sessions.get(&id)
+        self.sessions.lock().unwrap_or_else(|_| std::process::exit(1))
+    }
+    
+    /// ## Sets session's `source_path` field.
+    /// ----
+    /// Args:
+    /// ---
+    /// * `session_id` - id of the session to modify;
+    /// * `sourse_path` - new field value.
+    pub fn set_source_file(&self, session_id: &u128, source_path: &Path)
```

Comment:
in the future, this path may point not only to the current host but also to some external storage
is it possible to allow the path to be a URL string, for example:
```
enum SourcePath {
    FsPath(Path),
    UrlPath(String),
}
```

Reply:
Sure, will do. Haven't thought of that.